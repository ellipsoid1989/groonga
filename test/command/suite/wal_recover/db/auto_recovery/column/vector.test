# _database_close/_database_reopen can't use with HTTP because requests via HTTP are always
# processed by workers. Workers never use primary WAL role.
#@require-interface stdio

#@require-feature message_pack

#$GROONGA_ALLOW_DATABASE_REOPEN=yes
#$GRN_WAL_ROLE=primary

table_create Data TABLE_PAT_KEY ShortText
column_create Data tags COLUMN_VECTOR ShortText

load --table Data
[
{"_key": "abc",    "tags": ["A", "B", "C"]},
{"_key": "ABC",    "tags": ["B", "C", "D"]},
{"_key": "abcdef", "tags": ["F"]}
]

delete --table Data --key ABC

dump

_database_close

#@copy-path #{db_path}.0000101 #{db_path}.0000101.wal
#@add-important-log-levels notice
_database_reopen
#@remove-important-log-levels notice

dump
