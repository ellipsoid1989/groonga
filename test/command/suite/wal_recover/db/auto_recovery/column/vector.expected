table_create Data TABLE_PAT_KEY ShortText
[[0,0.0,0.0],true]
column_create Data tags COLUMN_VECTOR ShortText
[[0,0.0,0.0],true]
load --table Data
[
{"_key": "abc",    "tags": ["A", "B", "C"]},
{"_key": "ABC",    "tags": ["B", "C", "D"]},
{"_key": "abcdef", "tags": ["F"]}
]
[[0,0.0,0.0],3]
delete --table Data --key ABC
[[0,0.0,0.0],true]
dump
table_create Data TABLE_PAT_KEY ShortText
column_create Data tags COLUMN_VECTOR ShortText

load --table Data
[
["_key","tags"],
["abc",["A","B","C"]],
["abcdef",["F"]]
]
_database_close
[[0,0.0,0.0],true]
_database_reopen
[[0,0.0,0.0],true]
#|e| [reader][read-entry][ja][recover][Data.tags] must be map: <positive-integer>(2): path:<db/db.0000101>
#|n| [db][wal][recover] rebuild broken column: <Data.tags>(257)
#|n| DDL:258:column_create Data #recovering#tags
#|n| spec:258:update:Data.#recovering#tags:65(column:var_size):14(ShortText)
#|n| DDL:257:obj_remove Data.tags
#|n| spec:257:remove:Data.tags:65(column:var_size)
#|n| [db][wal][recover] succeeded to rebuild broken column: <Data.tags>(258)
dump
table_create Data TABLE_PAT_KEY ShortText
column_create Data tags COLUMN_VECTOR ShortText

load --table Data
[
["_key","tags"],
["abc",["A","B","C"]],
["abcdef",["F"]]
]
